### PR TITLE
fix(api): align master data routes with skills and add documentation

### PR DESCRIPTION
**Pull request by ashleythedeveloper**
This PR audits the master data API routes (organisations, facilities, products), fixes all gaps found, and adds documentation to the documentation site.

## Changes

**Route fixes (6 route files, 3 test files, 1 schema file):**
- Remove `tenantId` from all log metadata (auto-injected by `withTenantAuth` via `AsyncLocalStorage`)
- Align log message phrasing with established routes (query filters, nested filter metadata, count + total on list success)
- Register `Product`, `Organisation`, and `Facility` Zod schemas in `generateOpenAPISchemas()` with `secondaryIdentifierIds`
- Add missing Swagger fields: POST request body optional properties, DELETE 400 for products, POST 404 for organisations, fix organisation PATCH location type from `string` to `object`
- Fix Swagger search descriptions to reflect actual behaviour (name + identifier value)
- Correct "brand organisation" to "producing organisation" terminology
- Update DELETE handlers to use `new NextResponse(null, { status: 204 })` with `MockNextResponse` class in test mocks
- Strip unknown fields in all three PATCH handlers to only pass known updatable fields to the repository
- Align facilities test mock to use `handleRouteError` (matching products and organisations pattern)

**E2E tests (3 test files):**
- Add `no ok property` assertions on list endpoints
- Enhance pagination tests with `hasMore`, `limit`, `offset`, `total` assertions
- Add PATCH validation tests (empty body, invalid name, nonexistent ID)
- Add DELETE nonexistent ID tests
- Add invalid JSON body tests
- Add nonexistent foreign key reference tests (parentId, primaryIdentifierId, operatingOrganisationId)
- Add `response.body.error` assertions to all error paths

**Documentation (5 new files, 3 sidebar position bumps):**
- Add master data conceptual overview with entity relationship diagram, identifier model, location structure, and credential linking flow
- Add API reference pages for organisations, facilities, and products
- Document SetNull cascade behaviour on organisation and facility deletion
- Bump sidebar positions for API, Verify Page, and Operations sections

## Test plan
- [x] All 876 unit tests pass (`yarn jest --maxWorkers=1` from RI directory)
- [x] All 6 master data route test suites pass
- [x] Documentation site builds and renders correctly on localhost:3030

### CODE REVIEW

File: packages/reference-implementation/src/app/api/v1/facilities/[id]/route.ts [hunk 1]
```diff
@@ -170,10 +170,17 @@ export const PATCH = withTenantAuth(async (req, { tenantId, params }) => {
...
-  const updated = await updateFacility(id, tenantId, body);
+  const updateData: Record<string, unknown> = {};
+  for (const field of UPDATABLE_FIELDS) {
+    if (field in body) {
+      updateData[field] = body[field];
+    }
+  }
+
+  logger.info({ facilityId: id }, 'Updating facility');
+  const updated = await updateFacility(id, tenantId, updateData);
```
[the remark this answers is not in the document]
Comment (ashleythedeveloper) [reply packages/reference-implementation/src/app/api/v1/facilities/[id]/route.ts]: Done in 6d952a0e — added UpdateFacilityInput import and type cast.

File: packages/reference-implementation/src/app/api/v1/products/[id]/route.ts [hunk 2]
```diff
@@ -162,23 +162,28 @@ export const PATCH = withTenantAuth(async (req, { tenantId, params }) => {
...
+  // Pick only known updatable fields (level is immutable and silently excluded)
+  const updateData: Record<string, unknown> = {};
+  for (const field of UPDATABLE_FIELDS) {
+    if (field in body) {
+      updateData[field] = body[field];
+    }
+  }
+
+  logger.info({ productId: id }, 'Updating product');
+  const updated = await updateProduct(id, tenantId, updateData);
```
[the remark this answers is not in the document]
Comment (ashleythedeveloper) [reply packages/reference-implementation/src/app/api/v1/products/[id]/route.ts]: Done in 6d952a0e — added UpdateProductInput import and type cast.
